Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -91,59 +91,58 @@ @interface Branch() <BNCDebugConnectionDelegate, UIGestureRecognizerDelegate, BN
 
 @implementation Branch
 
-static Branch *currInstance;
-
 #pragma mark - Public methods
 
 
 #pragma mark - GetInstance methods
 
 + (Branch *)getInstance {
-    return [Branch getBranchInstance:YES];
+    NSString *branchKey = [BNCPreferenceHelper getBranchKey:YES];
+    if (!branchKey || [branchKey isEqualToString:NO_STRING_VALUE]) {
+        NSLog(@"Branch Warning: Please enter your branch_key in the plist!");
+    }
+
+    return [Branch getInstanceInternal];
 }
 
 + (Branch *)getTestInstance {
-    return [Branch getBranchInstance:NO];
+    NSString *branchKey = [BNCPreferenceHelper getBranchKey:NO];
+    if (!branchKey || [branchKey isEqualToString:NO_STRING_VALUE]) {
```

Comment:
I don't think that we should check app key here, though, because we don't have a concept of "test" app keys.